File: src/test/java/io/georocket/util/JsonUtilsTest.java
package io.georocket.util;

import io.vertx.core.json.JsonObject;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Tests for {@link JsonUtils}
 * @author Michel Kraemer
 */
public class JsonUtilsTest {
  /**
   * Test if a simple object can be merged
   */
  @Test
  public void simple() {
    JsonObject obj = new JsonObject()
        .put("type", "Person")
        .put("person", new JsonObject()
            .put("firstName", "Clifford")
            .put("lastName", "Thompson")
            .put("age", 40)
            .put("address", new JsonObject()
                .put("street", "First Street")
                .put("number", 6550)));
    JsonObject expected = new JsonObject()
        .put("type", "Person")
        .put("person.firstName", "Clifford")
        .put("person.lastName", "Thompson")
        .put("person.age", 40)
        .put("person.address.street", "First Street")
        .put("person.address.number", 6550);
    assertEquals(expected, JsonUtils.flatten(obj));
  }
}

File: src/test/kotlin/io/georocket/util/JsonUtilsTest.kt
package io.georocket.util

import io.georocket.util.JsonUtils.flatten
import io.vertx.kotlin.core.json.jsonObjectOf
import org.junit.Assert
import org.junit.Test

/**
 * Tests for [JsonUtils]
 * @author Michel Kraemer
 */
class JsonUtilsTest {
  /**
   * Test if a simple object can be merged
   */
  @Test
  fun simple() {
    val obj = jsonObjectOf(
      "type" to "Person",
      "person" to jsonObjectOf(
        "firstName" to "Clifford",
        "lastName" to "Thompson",
        "age" to 40,
        "address" to jsonObjectOf(
          "street" to "First Street",
          "number" to 6550
        )
      )
    )
    val expected = jsonObjectOf(
      "type" to "Person",
      "person.firstName" to "Clifford",
      "person.lastName" to "Thompson",
      "person.age" to 40,
      "person.address.street" to "First Street",
      "person.address.number" to 6550,
    )
    Assert.assertEquals(expected, flatten(obj))
  }
}
